test/clj/rems/test_browser.clj

```diff
@@ -236,3 +236,36 @@
     ;; if there is a js exception, nothing renders, so let's check
     ;; that we have lots of examples in the dom:
     (is (< 60 (count (query-all *driver* {:class :example}))))))
+
+(deftest test-language-change
+  (with-postmortem *driver* {:dir reporting-dir}
+    (testing "default language is English"
+      (go *driver* +test-url+)
+      (wait-visible *driver* {:tag :h1 :fn/text "Welcome to REMS"})
+      (login-as "alice")
+      (wait-visible *driver* {:tag :h1, :fn/text "Catalogue"})
+      (wait-page-loaded))
+
+    (testing "changing language while logged out"
+      (click *driver* :logout)
+      (wait-visible *driver* {:tag :h1 :fn/text "Welcome to REMS"})
+      (click *driver* [{:fn/has-class :btn-link :fn/text "FI"}])
```

Comment:
This should look for the button under `.language-switcher` to be explicit that it's a link for changing the language.

Reply:
Using `.language-switcher` is indeed good. It could be YAGNI for the extraction.

Reply:
I extracted `change-language` and `logout` just for slight improvement in readability.